src/components/sbb-tag/sbb-tag.tsx

```diff
@@ -55,56 +59,90 @@ export class SbbTag implements ComponentInterface, AccessibilityProperties {
+    }
+  }
+
+  @Watch('value')
+  public handleValueChange(currentValue: string, previousValue: string): void {
+    if (this.checked && currentValue !== previousValue) {
+      this.stateChange.emit({ type: 'value', value: currentValue });
+    }
+  }
+
+  /**
+   * Internal event that emits whenever the state of the tag
+   * in relation to the parent toggle changes.
+   */
+  @Event({
+    bubbles: true,
+    eventName: 'state-change',
+  })
+  public stateChange: EventEmitter<TagStateChange>;
+
+  /** Input event emitter */
+  @Event({ bubbles: true, composed: true }) public input: EventEmitter;
+
+  /** @deprecated only used for React. Will probably be removed once React 19 is available. */
+  @Event({ bubbles: true }) public didChange: EventEmitter;
+
+  /** Change event emitter */
+  @Event({ bubbles: true }) public change: EventEmitter;
+
   @Listen('sbbNamedSlotChange', { passive: true })
   public handleSlotNameChange(event: CustomEvent<Set<string>>): void {
     this._namedSlots = queryNamedSlotState(this._element, this._namedSlots, event.detail);
   }
 
   @Listen('click')
   public handleClick(event: Event): void {
-    forwardHostEvent(event, this._element, this._checkbox);
+    forwardHostEvent(event, this._element, this._button);
   }
 
-  /**
-   * @deprecated only used for React. Will probably be removed once React 19 is available.
-   */
-  @Event({ bubbles: true, cancelable: true }) public didChange: EventEmitter;
-
-  private _checkbox: HTMLInputElement;
+  private _button: HTMLButtonElement;
 
   public connectedCallback(): void {
     this._namedSlots = queryAndObserveNamedSlotState(this._element, this._namedSlots);
-    // Forward focus call to input element
-    this._element.focus = focusInputElement;
+    // Forward focus call to button element
+    this._element.focus = focusActionElement;
   }
 
-  /** Method triggered on checkbox change. Inverts the checked value and emits events. */
-  public checkedChanged(event: Event): void {
-    this.checked = this._checkbox?.checked;
-    forwardEventToHost(event, this._element);
+  /** Method triggered on button click. Inverts the checked value and emits events. */
+  public toggleClicked(): void {
+    const tagGroup = this._element.closest('sbb-tag-group') as HTMLSbbTagGroupElement;
+
+    // Prevent deactivating on exclusive / radio mode
+    if (tagGroup && !tagGroup.multiple && this.checked) {
+      return;
+    }
+    this.checked = !this.checked;
+    this.input.emit();
+    this.change.emit();
     this.didChange.emit();
   }
 
   public render(): JSX.Element {
     return (
-      <label class="sbb-tag__wrapper">
-        <input
-          ref={(checkbox: HTMLInputElement) => (this._checkbox = checkbox)}
-          type="checkbox"
+      <Host role="presentation">
+        <button
+          type="button"
+          class="sbb-tag"
+          // We have to ensure that the value is always present
+          aria-pressed={this.checked ? 'true' : 'false'}
```

Comment:
```suggestion
          aria-pressed={String(this.checked)}
```
could it work?

Reply:
thanks! changed to toString() as found in other examples in lyne. But your suggestion would work as well :)